javascript/src/stores/streams/StreamsStore.ts

```diff
@@ -16,16 +14,116 @@ interface Stream {
     createdAt: number;
 }
 
-var StreamsStore = {
-    listStreams(): JQueryPromise<Array<Stream>> {
-        var url = jsRoutes.controllers.api.StreamsApiController.listStreams().url;
-        var promise = $.getJSON(url);
-        promise.fail((jqXHR, textStatus, errorThrown) => {
-            UserNotification.error("Loading streams failed with status: " + errorThrown,
-                "Could not load streams.");
-        });
-        return promise;
-    },
+interface TestMatchResponse {
+    matches: boolean;
+    rules: any;
+}
+
+interface Callback {
+    (): void;
+}
+
+class StreamsStore {
+    private callbacks: Array<Callback> = [];
+
+    listStreams() {
+        return $.getJSON(jsRoutes.controllers.api.StreamsApiController.list().url);
```

Comment:
This method is still being used on `javascript/src/components/search/MessageDetail.jsx` and will not alert the user when the streams could not be loaded for some reason. Is this really what we want?